src/python_testing/TC_IDM_4_2.py

```diff
@@ -0,0 +1,170 @@
+        
+        CR1_ace_full_access = Clusters.AccessControl.Structs.AccessControlEntryStruct(
+            privilege=Clusters.AccessControl.Enums.AccessControlEntryPrivilegeEnum.kAdminister,
+            authMode=Clusters.AccessControl.Enums.AccessControlEntryAuthModeEnum.kCase,
+            subjects=[CR1_node_id],
+            targets=[])
+        
+        acl = [CR1_ace_full_access]
+        await self.write_acl(acl)
+        
+        # Read ServerList attribute
+        self.print_step("0a", "CR1 reads the Descriptor cluster ServerList attribute from EP0")
+        ep0_servers = await self.read_descriptor_server_list_expect_success(CR1)
+        
+        # Check if ep0_servers contains the ICD Management cluster ID (0x0046)
+        if Clusters.IcdManagement.id in ep0_servers:
+            # Read the IdleModeDuration attribute from the DUT
+            self.print_step("0b", "CR1 reads from the DUT the IdleModeDuration attribute and sets SUBSCRIPTION_MAX_INTERVAL_PUBLISHER_LIMIT = IdleModeDuration")
+            logging.info("debux - Setting IcdManagement cluster...")
+            cluster = Clusters.Objects.IcdManagement
+            
+            logging.info("debux - Setting IdleModeDuration attribute...")
+            idle_mode_duration_attr = Clusters.IcdManagement.Attributes.IdleModeDuration
+        
+            logging.info("debux - Getting idleModeDuration attribute value...")
+            idleModeDuration = await self.read_single_attribute_check_success(
+                cluster=cluster, 
+                attribute=idle_mode_duration_attr, 
+                dev_ctrl=CR1, 
+                node_id=self.dut_node_id, 
+                endpoint=0)
+            
+            logging.info('debux - idleModeDuration: ' + idleModeDuration)
+            
+            SUBSCRIPTION_MAX_INTERVAL_PUBLISHER_LIMIT = idleModeDuration
+        else:
+            # Defaulting SUBSCRIPTION_MAX_INTERVAL_PUBLISHER_LIMIT to 60 minutes
+            self.print_step("0b", "Set SUBSCRIPTION_MAX_INTERVAL_PUBLISHER_LIMIT = 60 mins")
+            SUBSCRIPTION_MAX_INTERVAL_PUBLISHER_LIMIT = 3600
+        
+        # Step 1
+        self.print_step(1, "CR1 sends a subscription message to the DUT with MaxIntervalCeiling set to a value greater than SUBSCRIPTION_MAX_INTERVAL_PUBLISHER_LIMIT. DUT sends a report data action to the TH. CR1 sends a success status response to the DUT. DUT sends a Subscribe Response Message to the CR1 to activate the subscription.")
+        min_interval_floor_sec = SUBSCRIPTION_MAX_INTERVAL_PUBLISHER_LIMIT - 60
+        max_interval_ceiling_sec = SUBSCRIPTION_MAX_INTERVAL_PUBLISHER_LIMIT + 60
+        
+        sub = await CR1.ReadAttribute(
+            nodeid=self.dut_node_id,
+            attributes=[(0, Clusters.BasicInformation.Attributes.NodeLabel)],
+            reportInterval=(min_interval_floor_sec, max_interval_ceiling_sec),
+            keepSubscriptions=False
+        )
+        
+        output_queue = queue.Queue()
```

Comment:
Since you're not waiting for reports here, you don't need to set up an attribute change callback. You will want this callback mechanism for later steps, but for this step, you just need the intervals.